p2pnetwork/nodeconnection.py

```diff
@@ -187,38 +188,28 @@ def run(self):
         """The main loop of the thread to handle the connection with the node. Within the
            main loop the thread waits to receive data from the node. If data is received 
            the method node_message will be invoked of the main node to be processed."""
-        buffer = b''  # Hold the stream that comes in!
-
         while not self.terminate_flag.is_set():
-            chunk = b''
-
             try:
-                chunk = self.sock.recv(4096)
+                packet_length = self.sock.recv(8)
```

Comment:
How do you know that you read here the length of the message? If you read here 8 bytes containing json for example, you get a very large length back. Besides simply sending a length, we need to have some protocol or some frame definition. Making sure that the sending and recieving end know when the message has started. Frame example: START_BYTE (1), LENGTH (8), DATA (LENGTH), END_BYTE(1). Also a CRC or Hash could be added for verification. And if we do this and see that the message was wrong, we should ask to resend it? Maybe that is for know a bit too far. We need some protocol to also get resilient against mistakes or simple attacks.